src/pages/_app.js

```diff
@@ -1,29 +1,45 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { useRouter } from "next/router";
 import transactionsData from "@/src/data/transactions.json";
 import { interFont } from "@/src/styles/font";
 import "@/src/styles/global.css";
 import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
-  const [initialData, setInitialData] = useState(transactionsData);
+  const [initialData, setInitialData] = useState([]);
   const [mode, setMode] = useState("default");
   const [action, setAction] = useState("default");
   const router = useRouter();
 
-  function handleMode(newMode) {
-    setMode(newMode);
+  useEffect(() => {
+    const localData = localStorage.getItem("transactions");
+    if (localData) {
+      setInitialData(JSON.parse(localData));
+    } else {
+      localStorage.setItem("transactions", JSON.stringify(transactions));
+      setInitialData(transactions);
+    }
+  }, []);
+
+  function handleMode(mode) {
+    setMode(mode);
   }
 
   function handleAddTransaction(data) {
-    setInitialData([{ id: uid(), ...data, amount: parseFloat(data.amount) }, ...initialData]);
+    const newTransaction = {
+      id: uid(),
+      ...data,
+      amount: parseFloat(data.amount),
+    };
+    const updatedData = [newTransaction, ...initialData];
+    setInitialData(updatedData);
+    localStorage.setItem("transactions", JSON.stringify(updatedData));
   }
```

Comment:
No need to use setInitialData(updatedData) , instead it can be directly set as setTransactions([newTransaction, ...transaction])

Reply:
good point! thanks